### PR TITLE
bpo-37961: fix regression in tracemalloc.Traceback.__repr__

### PR DESCRIPTION
**Pull request by blueyed**
Regression in 8d59eb1b66c51b2b918da9881c57d07d08df43b7
(v3.8.0b1-1208-g8d59eb1b66).

https://bugs.python.org/issue37961

### CODE REVIEW

File: Lib/tracemalloc.py [hunk 1]
```diff
@@ -226,7 +226,7 @@ def __str__(self):
         return str(self[0])
 
     def __repr__(self):
-        s = "<Traceback %r" % tuple(self)
+        s = f"<Traceback {tuple(self)}"
```
Comment (jd) [inline Lib/tracemalloc.py:229]: I'm confused about what this changes?

**Comment by vstinner** [comment]
Can you please explain what is the bug? https://bugs.python.org/issue37961 is closed, I suggest to open a new issue to explain your bug.

File: Lib/tracemalloc.py (hunk 1 above)
Comment (blueyed) [reply Lib/tracemalloc.py:229]: `"%r" % (1,2,3)` raises: "TypeError: not all arguments converted during string formatting".
This was changed (apparently unintentionally) in https://github.com/python/cpython/commit/8d59eb1b66c51b2b918da9881c57d07d08df43b7#diff-cbaacb7ac396956c9595b4899ef2ab31b02b264099e02eba56fbe2f5f2e441c0L224-R229.

**Comment by blueyed** [comment]
@vstinner 
> Can you please explain what is the bug? https://bugs.python.org/issue37961 is closed, I suggest to open a new issue to explain your bug.

I wanted to avoid noise in the issue tracker, but can do so later if you still think it is not a fixup/follow-up to the closed issue.

File: Lib/tracemalloc.py (hunk 1 above)
Comment (jd) [reply Lib/tracemalloc.py:229]: nice catch

**Review by jd — APPROVED** [review]

**Comment by vstinner** [comment]
The 3.8 branch is not affected:

```
        return "<Traceback %r>" % (tuple(self),)
```

**Comment by vstinner** [comment]
Well spotted, thanks for the fix, especially for additional tests! I merged your PR and backported it to 3.9.

**Pull request merged**